### PR TITLE
groups: Add code-organization group

### PR DESCRIPTION
**Pull request by navidshaikh**
for managing snyk scan CI job secrets

/cc: @spiffxp @dims 
xref: 
- https://github.com/kubernetes/kubernetes/issues/101528
- https://kubernetes.slack.com/archives/CHGFYJVAN/p1621848970007300

### CODE REVIEW

File: groups/groups.yaml [hunk 1]
```diff
@@ -1446,3 +1446,13 @@ groups:
...
+
+  - email-id: k8s-infra-code-organization@kubernetes.io
+    name: k8s-infra-code-organization
+    description: |-
+      ACL for managing snyk CI job secrets under code-organization
+    settings:
+      ReconcileMembers: "true"
+    members:
+      - davanum@gmail.com
+      - shaikhnavid14@gmail.com
```
Comment (navidshaikh) [inline groups/groups.yaml:1458]: /hold

@dims: should I add more folks here ?

Comment (dims) [reply groups/groups.yaml:1458]: Please add @liggitt and @BenTheElder at least. we can add others over time.

**Review by spiffxp — APPROVED** [review]
/approve
/lgtm

**Comment by spiffxp** [comment]
I'll open a followup PR to give this group permissions to admin the snyk secret

**Comment by BenTheElder** [comment]
thanks!

**Pull request merged**